subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DefaultDependencyManagementServices.java

```diff
@@ -202,6 +200,7 @@ public DependencyResolutionScopeServices(DomainObjectContext domainObjectContext
 
         void configure(ServiceRegistration registration) {
             registration.add(DefaultTransformedVariantFactory.class);
+            registration.add(DefaultConfigurationFactory.class);
```

Comment:
Any need to mark the new factory class @ThreadSafe like the other factory here?

Reply:
Sure, why not. The class should be thread safe, it's just a factory.

Reply:
Done!